x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_lifecycle/hooks/use_ingestion_rate.tsx

```diff
@@ -143,10 +171,11 @@ export const useIngestionRatePerTier = ({
 
       const start = datemath.parse(timeRange.from);
       const end = datemath.parse(timeRange.to);
-      const interval = getCalculateAutoTimeExpression((key) => core.uiSettings.get(key))(timeRange);
-      if (!start || !end || !interval) {
+      const intervalData = getIntervalAndType(timeRange, core);
```

Comment:
The overview page now uses [computeInterval](https://github.com/elastic/kibana/pull/214196#discussion_r2003393806) from viz-utils, should we align and use it here as well ?

Reply:
I think that won't work for aggs-based data fetching. Alternatively we could rewrite the data fetching here in ESQL, wdyt @klacabane ?

Reply:
My bad didn't notice we can't reuse it for dsl query. I think aligning on esql would be a good idea, but better tackled separately :)

Reply:
This rewrite would also require us to resolve the sampling thing (since that is not supported in ESQL yet AFAIK)